eng/scripts/Create-APIView.ps1

```diff
@@ -86,6 +86,32 @@ function Get-ResourceProviderFromReadMePath {
     return $null
 }
 
+function Get-ImpactedTypespecProjects {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$TypeSpecFile
+    )
+    $filePathParts = $TypeSpecFile.split([IO.Path]::DirectorySeparatorChar)
+    while ($filePathParts.Length -and !$configFilesInTypeSpecProjects) {
+      $filePathParts = $filePathParts | Select-Object -SkipLast 1
+      $typeSpecProjectBaseDirectory = $filePathParts -join [IO.Path]::DirectorySeparatorChar
+      $configFilesInTypeSpecProjects = Get-ChildItem -Path $typeSpecProjectBaseDirectory -File "tspconfig.yaml"
```

Comment:
This will still be an issue. It is possible to have tspconfig.yaml in sub folders for a shared project. Current logic will stop looking further upwards and move to next block to check if current directory has main.tsp. It won't come back and check again the parent path of current file path if there is no main.tsp.

We can simplify this logic that will work for shared projects also.

for each changed files, strip the prefix before `specification/' from path:
 1. check if current path has tspconfig.yaml and main.tsp.
 2. If yes, add this folder to project set
 3. If no, then go upwards and continue step 2 until we match current path to 'specification\/[^\/]+\/+`

APIView needs to be generated from paths that contain both main.tsp and tspconfig.yaml (at least to retain same behavior as current one.

Reply:
Hmm I don't think I understand the exact situation where this might not work.
The code right now is checking all parent directory of the changed file till we reach root.
line 96 `$filePathParts = $filePathParts | Select-Object -SkipLast 1` deletes the leaf directory on each iteration